### ISSUE TITLE
tslint trying to lint JSON

### ISSUE THREAD

**Issue by JustinBeckwith** [google/ts-style#172]
I'm trying to use the new [JSON imports](https://blogs.msdn.microsoft.com/typescript/2018/05/31/announcing-typescript-2-9/#json-imports) in TypeScript 2.9.  I'm using a simple import for a JSON file:

```js
import cfg from 'operations_client_config.json';
```

It builds great, but is failing at lint time with an error.  Not sure why tslint is all of the sudden deciding that it should import the JSON file and lint it as TypeScript 🤷‍♂️ 

```
> google-gax@0.16.1 check /Users/beckwith/Code/gax-nodejs
> gts check

The 'no-reference' rule threw an error in '/Users/beckwith/Code/gax-nodejs/src/operations_client_config.json':
TypeError: Cannot read property 'length' of undefined
    at walk (/Users/beckwith/Code/gax-nodejs/node_modules/tslint/lib/rules/noReferenceRule.js:46:67)
    at Rule.AbstractRule.applyWithFunction (/Users/beckwith/Code/gax-nodejs/node_modules/tslint/lib/language/rule/abstractRule.js:39:9)
    at Rule.apply (/Users/beckwith/Code/gax-nodejs/node_modules/tslint/lib/rules/noReferenceRule.js:27:21)
    at Linter.applyRule (/Users/beckwith/Code/gax-nodejs/node_modules/tslint/lib/linter.js:197:29)
    at /Users/beckwith/Code/gax-nodejs/node_modules/tslint/lib/linter.js:139:85
    at Object.flatMap (/Users/beckwith/Code/gax-nodejs/node_modules/tslint/lib/utils.js:151:29)
    at Linter.getAllFailures (/Users/beckwith/Code/gax-nodejs/node_modules/tslint/lib/linter.js:139:32)
    at Linter.lint (/Users/beckwith/Code/gax-nodejs/node_modules/tslint/lib/linter.js:99:33)
    at srcFiles.forEach.file (/Users/beckwith/Code/gax-nodejs/node_modules/gts/build/src/lint.js:43:20)
    at Array.forEach (<anonymous>)
/Users/beckwith/Code/gax-nodejs/src/operations_client_config.json
unused expression, expected an assignment or function call (no-unused-expression)
> 1 | {
  2 |   "interfaces": {
  3 |     "google.longrunning.Operations": {
  4 |       "retry_codes": {

Missing semicolon (semicolon)
  44 |     }
  45 |   }
> 46 | }
```

**Comment by ofrobots** [comment]
I believe this is the same root cause as #156. I'll look into these next.

**Comment by ofrobots** [comment]
Not related to #156.

**Comment by ofrobots** [comment]
Can you give me a repro, or a failing test case. Here's what I am trying right now (with TS 2.9):

```ts
test.serial('should handle json files correctly resolveJsonModule', t => {
  return withFixtures({
    'tsconfig.json': JSON.stringify({
      include: ['src'],
      compilerOptions: {
        module: 'commonjs',
        resolveJsonModule: true,
        esModuleInterop: true
      }
    }),
    src: {
      'a.ts': GOOD_CODE,
      'test.json': JSON.stringify({
        dry: false,
        debug: false
      })
    }
  }, async () => {
    const okay = lint.lint(OPTIONS);
    t.true(okay);
  });
});
```

The test passes.

**Comment by ofrobots** [comment]
Nevermind. Here's a failing test:

```ts
test.serial('should handle json files correctly resolveJsonModule', t => {
  return withFixtures({
    'tsconfig.json': JSON.stringify({
      include: ['src'],
      compilerOptions: {
        module: 'commonjs',
        resolveJsonModule: true,
        esModuleInterop: true
      }
    }),
    src: {
      'a.ts': `import settings from "./test.json";`,
      'test.json': JSON.stringify({
        dry: false,
        debug: false
      })
    }
  }, async () => {
    const okay = lint.lint(OPTIONS);
    t.true(okay);
  });
});
```

**Comment by ofrobots** [comment]
https://github.com/palantir/tslint/issues/3957

**Comment by stevenvolckaert** [comment]
We have the same problem in a greenfield project.

In our case, we had to disable the rule `no-unused-expression` in the `tsconfig.json` file for the entire project. I'm new to TSLint, and was looking for ways to disable this rule, just for `*.json` files.

The only way I can think of. is to add

```
/* tslint:disable:no-unused-expression */
```

at the top of each JSON file I'd like to import with an ES6 `import` statement.

**However, this is impossible as no comments are allowed in valid JSON.** Adding the comment invalidates the JSON file...

We're waiting for a solution. In the meantime, perhaps there's a workaround for this? (Except disabling the `no-unused-expression` rule for the entire project.)

**Comment by GabrielDelepine** [comment]
Please add in your `tslint.json` file

``` 
{
  "linterOptions": {
    "exclude": [
      "*.json",
      "**/*.json"
    ]
  }
}
```

`**/*.json` is necessary to be recursive

Available since version `5.8`

**Comment by GabrielDelepine** [comment]
Please add in your `tslint.json` file

``` 
{
  "linterOptions": {
    "exclude": [
      "*.json",
      "**/*.json"
    ]
  }
}
```

`**/*.json` is necessary to be recursive

Available since version `5.8`

**Comment by focux** [comment]
It worked like a charm! Thanks @GabrielDelepine!
